pkg/apis/core/validation/validation.go

```diff
@@ -3938,6 +3938,8 @@ func ValidateHostAliases(hostAliases []core.HostAlias, fldPath *field.Path) fiel
 // ValidateTolerations tests if given tolerations have valid data.
 func ValidateTolerations(tolerations []core.Toleration, fldPath *field.Path) field.ErrorList {
 	allErrors := field.ErrorList{}
+
+	uniqueTaints := map[core.TaintEffect]sets.Set[string]{}
```

Comment:
I guess using `map[string]core.TaintEffect` would seem simpler?

Reply:
need check `Effect` of `Toleration`, there is no need  convert to `string`.  need check `Key` of  `Toleration`, there is no need use `core.TaintEffect`.

Reply:
`core.Toleration` is unique with `Toleration.Key` and `core.TaintEffect`, one `core.TaintEffect` must have different `Toleration.Key`, so need use `core.TaintEffect` as key, `Toleration.Key` as values.